src/jit/gentree.h

```diff
@@ -574,11 +574,21 @@ struct GenTree
         return isContained() && IsCnsIntOrI() && !isUsedFromSpillTemp();
     }
 
+#if FEATURE_X87_DOUBLES
```

Comment:
When you say cost of upcast/downcast done during jitting?  I am not sure that is a significant enough cost to worry about.  

Do you have data to conclude the the two points in your reply?